superset/connectors/sqla/models.py

```diff
@@ -404,9 +404,14 @@ class SqlMetric(Model, BaseMetric):
     )
     export_parent = "table"
 
+    def get_template_processor(self, **kwargs: Any) -> BaseTemplateProcessor:
+        return self.table.get_template_processor()
+
     def get_sqla_col(self, label: Optional[str] = None) -> Column:
         label = label or self.metric_name
-        sqla_col: ColumnClause = literal_column(self.expression)
+        sqla_col: ColumnClause = literal_column(
+            self.get_template_processor().process_template(self.expression)
+        )
```

Comment:
It is not recommended to add the new instance method(get_template_processor) here.

```suggestion
    def get_sqla_col(self, label: Optional[str] = None) -> Column:
        label = label or self.metric_name

        tp = self.table.get_template_processor()
        sqla_col = literal_column(tp.process_template(self.expression))
        return self.table.make_sqla_column_compatible(sqla_col, label)
```

Reply:
Got it, 

Fixed the code, 

Why it is not recommended to have new instance method? I am asking only yo expand my knowledge

Thanks

Reply:
Hi Guyt. The main reason is no more `self.get_template_processor` method call currently, so no need to extend this class. If there is a need for this in the future, we can easily extend it.